agent/agent.go

```diff
@@ -427,34 +428,59 @@ func (a *Agent) handlePolicy(p *policystorage.Policy) {
 		// plugins doing this.
 		action.Canonicalize()
 
-		if !action.IsWithinLimits(p.Min, p.Max) {
+		// If the policy is configured with dry-run:true then we set the
+		// action count to nil so its no-nop. This allows us to still
+		// submit the job, but not alter its state.
+		if val, ok := p.Target.Config["dry-run"]; ok && val == "true" {
+			logger.Info("scaling dry-run is enabled, using no-op task group count",
+				"target_config", p.Target.Config)
+			action.Count = nil
```

Comment:
^^^ like, for example, logging the count that we would have set if dry-run hadn't been enabled.